### PR TITLE
Use new go-connlimit with HTTP 429 response

### PR DESCRIPTION
**Pull request by benbuzbee**
This is essentially a port of Consul's similar fix
Changes are:
go get -u github.com/hashicorp/go-connlimit
go mod vendor
Use new HTTP429 handler

https://github.com/hashicorp/consul/commit/20d1ea7d2d272b0d25a9b5af6ddec8748a3928aa

### CODE REVIEW

**Comment by benbuzbee** [comment]
@notnoop @tgross sorry for the random pings but I was hoping this could get a looksee since i think it's pretty low risk and would help us triage issues in our API clients easier

File: vendor/github.com/hashicorp/nomad/api/event_sink.go [hunk 1]
```diff
@@ -52,6 +52,6 @@ func (e *EventSinks) Register(eventSink *EventSink, w *WriteOptions) (*WriteMeta
 	return wm, nil
 }
 
-func (e *EventSinks) Deregister(name string, w *WriteOptions) (*WriteMeta, error) {
-	return e.client.delete("/v1/event/sink/"+name, nil, w)
+func (e *EventSinks) Deregister(id string, w *WriteOptions) (*WriteMeta, error) {
```
Comment (tgross) [inline vendor/github.com/hashicorp/nomad/api/event_sink.go:55]: We shouldn't have this floating around after a `make sync` at this point and it looks like there's a merge conflict... can you rebase on `master`?

**Review by tgross — CHANGES_REQUESTED** [review]
Hi @benbuzbee the dependency bump LGTM and the code changes look ok, but this appears to break some unit tests: https://app.circleci.com/pipelines/github/hashicorp/nomad/13189/workflows/855edac5-dbc2-48bb-b152-48e10681f9c8/jobs/120507

**Comment by shoenig** [comment]
Sorry @benbuzbee / @tgross I had started looking into this and forgot to comment - the unit test failures I believe are caused by the tests themselves being broken (in the HTTPS path, unintentionally only the timeout limiter is being exercised). I'll try to clean that up and merge this in soon if no-one beats me to it.

**Pull request closed without merging**